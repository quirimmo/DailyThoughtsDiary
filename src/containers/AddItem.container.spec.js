import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockAddItemThunk = jest
	.fn()
	.mockImplementation(() => ({ type: 'MOCKED_ACTION' }));
jest.mock('../actions/items.action', () => ({
	addItemThunk: item => mockAddItemThunk(item)
}));

jest.mock('moment', () => () => ({
	format: () => '2018–10–28T12:34:56+00:00'
}));

import AddItemContainer from './AddItem.container';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore();
let component;
let instance;

describe('AddItemContainer', () => {
	beforeEach(() => {
		const renderer = new ShallowRenderer();
		component = renderer.render(<AddItemContainer store={store} />);
		instance = renderer.getRenderOutput();
	});

	it('should be defined', () => {
		expect(AddItemContainer).toBeDefined();
	});

	it('should render the component', async () => {
		expect(component).toMatchSnapshot();
	});

	it('should define the addItem prop', () => {
		expect(typeof instance.props.addItem).toEqual('function');
	});

	it('should dispatch the addItemThunk action', () => {
		const item = {};
		instance.props.addItem(item);
		expect(mockAddItemThunk).toHaveBeenCalledWith(item);
	});
});
