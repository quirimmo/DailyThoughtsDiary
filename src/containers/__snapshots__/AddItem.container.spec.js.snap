// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItemContainer should render the component 1`] = `
<AddItemScreen
  addItem={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
